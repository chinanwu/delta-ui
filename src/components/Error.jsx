import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import getThemeClassname from '../functions/getThemeClassname';

import './Error.less';

export const Error = ({ dark }) => (
	<div
		className={getThemeClassname('Error', dark)}
		role="alert"
		aria-busy="true"
	>
		<div className="Error__content">
			<span>Something wrong happened! Please try again.</span>
			<Link to="/">
				<button id="errorHomeBtn" className="Error__homeBtn" role="link">
					Return Home
				</button>
			</Link>
		</div>
	</div>
);

export const mapStateToProps = ({ theme: { dark } }) => ({
	dark,
});

export default connect(mapStateToProps)(Error);

// TODO:
// - Dark mode

// Potential future improvements:
// - Make this into a toast?
