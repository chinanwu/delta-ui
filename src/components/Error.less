@import '../style/colours';
@import '../style/button';

.Error {
	position: fixed;
	width: 100%;
	top: 0;
	left: 0;
	right: 0;
	bottom: 0;
	display: flex;
	text-align: center;
	justify-content: center;
	align-items: center;
	font-size: 4rem;
	background-color: @modalBg;
}

.Error__content {
	background-color: white;
	width: 45rem;
	padding: 2rem;
	display: flex;
	flex-direction: column;
}

.Error__homeBtn {
	@largeBtn();
	margin-top: 2rem;
}
