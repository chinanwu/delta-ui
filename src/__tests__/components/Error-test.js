import { shallow } from 'enzyme';
import React from 'react';

import { Error } from '../../components/Error.jsx';

jest.unmock('../../components/Error.jsx');

describe('Error component', () => {
	describe('rendering', () => {
		it('renders Error', () => {
			const wrapper = shallow(<Error />);
			expect(wrapper.getElement()).toMatchSnapshot();
		});
	});
});
