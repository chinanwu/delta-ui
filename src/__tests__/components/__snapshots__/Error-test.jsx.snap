// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component rendering renders Error 1`] = `
<div
  aria-busy="true"
  className="Error"
  role="alert"
>
  <div
    className="Error__content"
  >
    <span>
      Something wrong happened! Please try again.
    </span>
    <ForwardRef
      to="/"
    >
      <button
        className="Error__homeBtn"
        id="errorHomeBtn"
        role="link"
      >
        Return Home
      </button>
    </ForwardRef>
  </div>
</div>
`;
